src/lib/mina/token/token-contract.ts

```diff
@@ -27,10 +29,31 @@ abstract class TokenContract extends SmartContract {
 
   deploy(args?: DeployArgs) {
     super.deploy(args);
+
+    // set access permission, to prevent unauthorized token operations
     this.account.permissions.set({
       ...Permissions.default(),
       access: Permissions.proofOrSignature(),
     });
+
+    // assert that this account is new, to ensure unauthorized token operations
+    // are not possible before this contract is deployed
+    // see https://github.com/o1-labs/o1js/issues/1439 for details
+    this.account.isNew.requireEquals(Bool(true));
```

Comment:
Is there a way we can provide documentation via intellisense on this rather than in code documentation? Is it even necessary to communicate this? Just spitting ideas here.

Reply:
I think this is a good idea! Maybe a doc-comment on `deploy()`, so that when you're going to override it by typing `super.deploy()`, you have documentation about what it does